indico/htdocs/js/indico/Core/Dialogs/Util.js

```diff
@@ -15,23 +15,64 @@
  * along with Indico; if not, see <http://www.gnu.org/licenses/>.
  */
 
-IndicoUI.Dialogs.Util = {
-    error: function(err) {
-        var dialog = null;
-        if (err.type === 'noReport') {
-            dialog = new NoReportErrorDialog(err);
-        } else {
-            dialog = new ErrorReportDialog(err);
-        }
-        dialog.open();
-    },
+(function() {
+    'use strict';
 
-    progress: function(text) {
-        var dialog = new ProgressDialog(text);
-        dialog.open();
+    var ERROR_DIALOG_TEMPLATE = _.template(
+        '<div class="error-dialog">' +
+        '    <div class="error-message-box">' +
+        '        <span class="icon"></span>' +
+        '        <div class="message-text"><%= errorText %></div>' +
+        '    </div>' +
+        '    <p class="js-info-text"><%= infoText %></p>' +
+        '    <div class="actions">' +
+        '        <button class="i-button big warning js-report-button"' +
+        '                data-href="<%= url %>"' +
+        '                data-title="<%= reportText %>"' +
+        '                data-ajax-dialog>' +
+        '            <%= reportText %>' +
+        '        </button>' +
+        '        <button class="i-button big" data-button-back><%= closeText %></button>' +
+        '    </div>' +
+        '</div>'
+    );
 
-        return function() {
-            dialog.close();
-        };
+    function showErrorDialog(error) {
+        var $content = $(ERROR_DIALOG_TEMPLATE({
+            errorText: error.message,
+            infoText: $T.gettext('Please reload the page and report this error to us if it persists.'),
+            reportText: $T.gettext('Report Error'),
+            closeText: $T.gettext('Close'),
+            url: error.report_url
+        }));
+        $content.on('ajaxDialog:closed', function(evt, data) {
+            if (data) {
+                $content.find('.js-report-button').prop('disabled', true);
+                $content.find('.js-info-text').text($T.gettext('Thanks for your error report!'));
+            }
+        });
+        ajaxDialog({
+            content: $content,
+            title: error.title,
+            onOpen: function() {
+                if (!error.report_url) {
+                    $content.find('.js-report-button').remove();
+                    $content.find('.js-info-text').remove();
+                }
+            }
+        });
     }
-};
+
+    IndicoUI.Dialogs.Util = {
+        error: showErrorDialog,
+
```

Comment:
Is this for readability or..?

Reply:
I think I'll just expose and call `showErrorDialog` directly instead of keeping the legacy `IndicoUI.Dialogs.Util.error` call